internal/cmd/operator-sdk/generate/internal/relatedimages.go

```diff
@@ -83,6 +85,28 @@ func (c *relatedImageCollector) collectFromEnvironment(containerRef string, env
 	return nil
 }
 
+func (c *relatedImageCollector) collectFromBaseCSV(base *operatorsv1alpha1.ClusterServiceVersion) error {
+	if base == nil || base.Spec.RelatedImages == nil {
+		return nil
+	}
+
+	for _, relatedImage := range base.Spec.RelatedImages {
+		c.collect(relatedImage.Name, relatedImage.Image, "baseCSV")
```

Comment:
I don't think we want to use `c.collect` here. That function has a bunch of logic to handle collisions, which we don't want applied to base CSVs. Base CSV related Images should just be moved over as is. I would think what we want is the final step of collecting related images is going through all the base CSV related images and for each:

1. If a discovererd related image has the same name and image: do nothing, we already have that an equivalent related image in the output.
2. If a discovererd related image has the same name, but different image: prefix the discovered related image name, put the base CSV related image in the output as is.
3. If a discovered related image has the same image, but different name: put the base CSV related image in the output as is. We typically don't want the same image listed twice, but in this case we accept it in the name of backwards compatibility.

All should have a clear warning that this is a deprecated approach, and will be removed in v2.